internal/store/postgres/user_repository.go

```diff
@@ -6,18 +6,31 @@ import (
 	"errors"
 	"fmt"
 	"strings"
+	"time"
 
 	"database/sql"
 
 	"github.com/doug-martin/goqu/v9"
+	"github.com/jmoiron/sqlx"
 	"github.com/odpf/shield/core/user"
 	"github.com/odpf/shield/pkg/db"
+	"github.com/odpf/shield/pkg/uuid"
 )
 
 type UserRepository struct {
 	dbc *db.Client
 }
 
+type userXMetadata struct {
```

Comment:
can you describe what is userXMetadata ?

Reply:
I defined this struct to store the rows from the join of the Users and Metadata table.
This could be handled by using both the User and Metadata struct together. But, that approach would be chaotic.

What do you think, which approach is better?

Reply:
changed struct to `joinUserMetadata`